src/Plugins/ComplexCore/src/ComplexCore/Filters/FindFeaturePhasesFilter.cpp

```diff
@@ -0,0 +1,146 @@
+                                                                const std::atomic_bool& shouldCancel) const
+{
+  auto pCellPhasesArrayPathValue = filterArgs.value<DataPath>(k_CellPhasesArrayPath_Key);
+  auto pFeatureIdsArrayPathValue = filterArgs.value<DataPath>(k_FeatureIdsArrayPath_Key);
+  auto pFeaturePhasesArrayPathValue = filterArgs.value<DataPath>(k_FeaturePhasesArrayPath_Key);
+
+  complex::Result<OutputActions> resultOutputActions;
+
+  const IDataArray& featureIds = dataStructure.getDataRefAs<IDataArray>(pFeatureIdsArrayPathValue);
+
+  {
+    auto createFeaturePhasesAction = std::make_unique<CreateArrayAction>(NumericType::int32, std::vector<usize>{featureIds.getNumberOfTuples()}, std::vector<usize>{1}, pFeaturePhasesArrayPathValue);
+    resultOutputActions.value().actions.push_back(std::move(createFeaturePhasesAction));
+  }
+
+  std::vector<PreflightValue> preflightUpdatedValues;
+
+  return {std::move(resultOutputActions), std::move(preflightUpdatedValues)};
+}
+
+//------------------------------------------------------------------------------
+Result<> FindFeaturePhasesFilter::executeImpl(DataStructure& dataStructure, const Arguments& filterArgs, const PipelineFilter* pipelineNode, const MessageHandler& messageHandler,
+                                              const std::atomic_bool& shouldCancel) const
+{
+  auto pCellPhasesArrayPathValue = filterArgs.value<DataPath>(k_CellPhasesArrayPath_Key);
+  auto pFeatureIdsArrayPathValue = filterArgs.value<DataPath>(k_FeatureIdsArrayPath_Key);
+  auto pFeaturePhasesArrayPathValue = filterArgs.value<DataPath>(k_FeaturePhasesArrayPath_Key);
+
+  const Int32Array& cellPhases = dataStructure.getDataRefAs<Int32Array>(pCellPhasesArrayPathValue);
+  const Int32Array& featureIds = dataStructure.getDataRefAs<Int32Array>(pFeatureIdsArrayPathValue);
+  Int32Array& featurePhases = dataStructure.getDataRefAs<Int32Array>(pFeaturePhasesArrayPathValue);
+
+  // Resize the featurePhases array to the proper size
+  usize featureIdsMaxIdx = std::distance(featureIds.begin(), std::max_element(featureIds.cbegin(), featureIds.cend()));
+  usize maxValue = featureIds[featureIdsMaxIdx];
+
+  DataStore<int32>& featurePhasesStore = featurePhases.getIDataStoreRefAs<DataStore<int32>>();
+  featurePhasesStore.reshapeTuples(std::vector<usize>{maxValue + 1});
+
+  usize totalPoints = featureIds.getNumberOfTuples();
+  std::map<int32, int32> featureMap;
+  std::map<int32, int32> warningMap;
+
+  for(usize i = 0; i < totalPoints; i++)
+  {
+    if(shouldCancel)
+    {
+      return {};
+    }
+
+    int32 gnum = featureIds[i];
+    if(featureMap.find(gnum) == featureMap.end())
+    {
+      featureMap.emplace(gnum, cellPhases[i]);
+    }
+
+    int32 curPhaseVal = featureMap[gnum];
+    if(curPhaseVal != cellPhases[i])
+    {
+      if(warningMap.find(gnum) == warningMap.end())
+      {
+        warningMap[gnum] = 1;
+      }
+      else
+      {
+        warningMap[gnum]++;
+      }
```

Comment:
```suggestion
      auto [iter, insertSuccess] = warningMap.insert({gnum, 1});
      if(!insertSuccess)
      {
        iter->second += 1;
      }
```
`insert`/`emplace` return a pair of the iterator to the inserted element or the already existing element and whether the insertion was successful. This way we can avoid a second lookup.